### ISSUE TITLE
Incorrectly named option 'rejectPublicSuffixes' in 'CookieJar'

### ISSUE THREAD

**Issue by yairmzr** [salesforce/tough-cookie#292]
### TL;DR
`rejectPublicSuffixes` should be renamed as setting it to `true` does the opposite of its name - it allows domains with public suffixes but rejects domains without a TLD.

### Explanation

The `rejectPublicSuffixes` is true by default.
It's explained in the Readme as: `reject cookies with domains like "com" and "co.uk"` [here](https://github.com/salesforce/tough-cookie/blob/v4.1.3/README.md#getpublicsuffixhostname:~:text=rejectPublicSuffixes%20%2D%20boolean%20%2D%20default%20true%20%2D%20reject%20cookies%20with%20domains%20like%20%22com%22%20and%20%22co.uk%22)

When `setCookie()` is called on a `CookieJar` instance that has this option enabled, the following code runs: https://github.com/salesforce/tough-cookie/blob/v4.1.3/lib/cookie.js#L1179

The code gets a `suffix` variable returned from the `getPublicSuffix()` function, and if it's `null` it would error with the message: `Cookie has domain set to a public suffix`.

The `getPublicSuffix()` function calls `psl.get(domain)` in the code here: https://github.com/salesforce/tough-cookie/blob/v4.1.3/lib/pubsuffix-psl.js#L70

The `psl.get()` function is documented in the `psl` package as returning the SLD+TLD for the specified domain: https://www.npmjs.com/package/psl#:~:text=psl.get(%27example.COM%27)%3B%20//%20%27example.com%27
such as:
```
psl.get('google'); // null
psl.get('google.com'); // 'google.com'
```

This behavior is also documented correctly [here](https://github.com/salesforce/tough-cookie/blob/v4.1.3/README.md#getpublicsuffixhostname:~:text=For%20example%3A%20www.example.com%20and%20www.subdomain.example.com%20both%20have%20public%20suffix%20example.com.).

Therefore, the `rejectPublicSuffixes` parameter does the exact opposite of its name and its documentation ( `reject cookies with domains like "com" and "co.uk"` ).
Instead of rejecting public suffixes, it allows all public suffixes (e.g. `google.com`) and rejects domains without a TLD (e.g. `example`).

Even though this may introduce compatibility-breaking changes - it's probably best to rename this parameter to reflect its purpose better and change its documentation accordingly.

**Comment by colincasey** [comment]
@yairmzr the domain `google.com` is not a public suffix domain.  This would be a valid domain to set a cookie with if `rejectPublicSuffixes` is set to `true`.  

Here is [how RFC 6265 defines a public suffix](https://datatracker.ietf.org/doc/html/rfc6265#section-5.3):
> NOTE: A "public suffix" is a domain that is controlled by a public registry, such as "com", "co.uk", and "pvt.k12.wy.us". This step is essential for preventing attacker.com from disrupting the integrity of example.com by setting a cookie with a Domain attribute of "com".

For example, this is valid:

```
const jar = new CookieJar(undefined, { rejectPublicSuffixes: true })
jar.setCookieSync('a=b; Domain=google.com; Path=/', 'https://google.com')
```

But this is not:

```js
const jar = new CookieJar(undefined, { rejectPublicSuffixes: true })
jar.setCookieSync('a=b; Domain=co.uk; Path=/', 'https://co.uk')
```
